### PR TITLE
Add Data Migration Overview

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Add Data Migration Overview

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/7480/files#diff-79e49474cab22cf17093196411c1ea1527c65786e8fc743b7460ba8abb288528
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by sunzhaoyang — APPROVED** [review]

File: data-migration/migration-overview.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+If cloud storage (S3) service is not used, the network connectivity is good, and the network latency is low, you can use the following method to migrate data from MySQL to TiDB.
+
+- [Migrate MySQL data of less than 1 TB to TiDB](/data-migration/migrate-mysql-tidb-less-tb.md)
+
+If you have a high demand on migration speed, or if the data size is large (for example, larger than 1 TiB), and you do not allow other applications to write to TiDB during the migration period, you can use TiDB Lightning to quickly import data. Then, you can use DM to replicate incremental data (binlog) based on your application needs.
+
+- [Migrate MySQL data of more than 1 TB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
+
+## Merge and migrate MySQL shards into TiDB
```
Comment (TomShawn) [inline data-migration/migration-overview.md:39]:
```suggestion
## Migrate and merge MySQL shards into TiDB
```

File: data-migration/migration-overview.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+If your application uses MySQL shards for data storage and you need to migrate these shards into TiDB as one table, you can use DM to perform the shard merge and migration.
+
+- [Merge and migrate MySQL shards of less than 1 TB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
```
Comment (TomShawn) [inline data-migration/migration-overview.md:43]:
```suggestion
- [Migrate and merge MySQL shards of less than 1 TB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
```

File: data-migration/migration-overview.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+If the data size of the sharded tables is large (for example, larger than 1 TiB), and you do not allow other applications to write to TiDB during the migration period, you can use TiDB Lightning to quickly merge and import the sharded tables. Then, you can use DM to replicate incremental sharding data (binlog) based on your application needs.
+
+- [Merge and migrate MySQL shards of more than 1 TB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
```
Comment (TomShawn) [inline data-migration/migration-overview.md:47]:
```suggestion
- [Migrate and merge MySQL shards of more than 1 TB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
```

File: data-migration/migration-overview.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+---
+title: Data Migration Overview
+summary: Learn the overview of data migration from databases or data formats (CSV/SQL).
```
Comment (sunzhaoyang) [inline data-migration/migration-overview.md:3]: data formats 感觉怪怪的....意思应该是得表达出从文件中导入吧？

File: data-migration/migration-overview.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+- [Migrate data from SQL files to TiDB](//data-migration/migrate-sql-file-tidb.md)
+
+## More complex migration solutions
+
+The following features can improve the migration process and might meet more needs in your application.
+
+- [Migrate with pt/gh-host](/data-migration/migrate-with-pt-ghost.md)
+- [Migrate with Binlog Event Filter](/data-migration/migrate-with-binlog-event-filter.md)
+- [Migrate with Filter Binlog Events Using SQL Expressions](/data-migration/migrate-with-binlog-sql-expression-filter.md)
+- [Migrate with More Columns in Downstream](//data-migration/migrate-with-more-columns-downstream.md)
```
Comment (sunzhaoyang) [inline data-migration/migration-overview.md:61]:
```suggestion
- [Migrate with More Columns in Downstream](/data-migration/migrate-with-more-columns-downstream.md)
```

**Review by sunzhaoyang — APPROVED** [review]
rest LGTM

File: data-migration/migration-overview.md (hunk 4 above)
Comment (TomShawn) [reply data-migration/migration-overview.md:3]:
```suggestion
summary: Learn the overview of data migration from databases or CSV/SQL files.
```

**Comment by TomShawn** [comment]
@sunzhaoyang Thanks~All comments are addressed.

File: data-migration/migration-overview.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+---
+title: Data Migration Overview
+summary: Learn the overview of data migration from databases or CSV/SQL files.
```
Comment (en-jin19) [inline data-migration/migration-overview.md:3]:
```suggestion
summary: Generally describes various data migration scenarios and corresponding data migration solutions
```

File: data-migration/migration-overview.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+If you have a high demand on migration speed, or if the data size is large (for example, larger than 1 TiB), and you do not allow other applications to write to TiDB during the migration period, you can use TiDB Lightning to quickly import data. Then, you can use DM to replicate incremental data (binlog) based on your application needs.
+
+- [Migrate MySQL data of more than 1 TB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
+
+## Migrate and merge MySQL shards into TiDB
+
+If your application uses MySQL shards for data storage and you need to migrate these shards into TiDB as one table, you can use DM to perform the shard merge and migration.
+
+- [Migrate and merge MySQL shards of less than 1 TB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
```
Comment (en-jin19) [inline data-migration/migration-overview.md:43]:
```suggestion
- [Migrate and merge MySQL shards of less than 1 TB to TiDB](/data-migration/migrate-sharding-mysql-tidb-less-tb.md)
```

File: data-migration/migration-overview.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+If the data size of the sharded tables is large (for example, larger than 1 TiB), and you do not allow other applications to write to TiDB during the migration period, you can use TiDB Lightning to quickly merge and import the sharded tables. Then, you can use DM to replicate incremental sharding data (binlog) based on your application needs.
+
+- [Migrate and merge MySQL shards of more than 1 TB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
```
Comment (en-jin19) [inline data-migration/migration-overview.md:47]:
```suggestion
- [Migrate and merge MySQL shards of more than 1 TB to TiDB](/data-migration/migrate-sharding-mysql-tidb-above-tb.md)
```

File: data-migration/migration-overview.md (hunk 6 above)
Comment (TomShawn) [reply data-migration/migration-overview.md:3]:
```suggestion
summary: Learn the overview of data migration scenarios and the solutions.
```

**Comment by TomShawn** [comment]
@en-jin19 All comments are addressed. PTAL again. Thanks!

File: data-migration/migration-overview.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+# Data Migration Overview
+
+This document gives an overview of the data migration solutions that you can use with TiDB. The data migration solutions are as follows:
+
+- Full data migration.
+    - To import Amazon Aurora snapshots, CSV files, or Mydumper SQL files into TiDB, you can use TiDB Lightning to perform the full migration.
+    - To export all TiDB data as CSV files or Mydumper SQL files, you can use Dumpling to perform the full migration, which makes data migration from MySQL or MariaDB easier.
+    - To migrate all data from a database with a total data size of less than 1 TB, you can also use TiDB Data Migration (DM).
```
Comment (en-jin19) [inline data-migration/migration-overview.md:13]:
```suggestion
    - To migrate all data from a database with a small data size volume (for example, less than 1 TiB), you can also use TiDB Data Migration (DM).
```

**Review by en-jin19 — APPROVED** [review]
LGTM

**Pull request merged**